### PR TITLE
Setting up Build for Lemur

### PR DESCRIPTION
**Pull request by mitchcail**
Setting up lemur to build through our gitlab CI

NOT INCLUDED AS PART OF THIS PR
- testing via gitlab will come in a subsequent PR, right now we are depending on github to run our tests

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+stages:
+  - build-stage-image
+  - build-prod-image
+
+build-stage-image:
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/images:$CURRENT_CI_IMAGE
+  stage:build-stage-image
+  when: on_success
+  # the 3h timeout is required for images compiling large projects for ARM using qemu
```
Comment (bobmshannon) [inline .gitlab-ci.yml:13]: Does this comment still apply to us? Not sure we are building an ARM compatible container image here.

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  # we can revert back to 1h when we use native builders for ARM
+  timeout: 3h
+  tags: ["arch:amd64"]
+  variables:
+    CI_ENABLE_CONTAINER_IMAGE_BUILDS: "true"
+  script:
+    - cd publish && docker buildx build --build-arg CI_COMMIT_SHA=${CI_COMMIT_SHA} --tag 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/lemur:$CI_COMMIT_TAG --push .
+  only:
+    - tags
```
Comment (bobmshannon) [inline .gitlab-ci.yml:36]: Ultra nit: missing newline at end of file

**Review by bobmshannon — APPROVED** [review]

File: .gitlab-ci.yml (hunk 2 above)
Comment (mitchcail) [reply .gitlab-ci.yml]: shoot, I had added newlines to those files but then forgot to push from my local branch. Files have been updated

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  stage:build-stage-image
+  when: on_success
+  # the 3h timeout is required for images compiling large projects for ARM using qemu
+  # we can revert back to 1h when we use native builders for ARM
+  timeout: 3h
+  tags: ["arch:amd64"]
+  variables:
+    CI_ENABLE_CONTAINER_IMAGE_BUILDS: "true"
+  script:
+    - cd publish&& docker buildx build --build-arg CI_COMMIT_SHA=${CI_COMMIT_SHA} --tag 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/lemur:v$CI_PIPELINE_ID-$CI_COMMIT_SHORT_SHA --push .
```
Comment (bobmshannon) [inline .gitlab-ci.yml:20]:
```suggestion
    - cd publish && docker buildx build --build-arg CI_COMMIT_SHA=${CI_COMMIT_SHA} --tag 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/lemur:v$CI_PIPELINE_ID-$CI_COMMIT_SHORT_SHA --push .
```

Comment (mitchcail) [reply .gitlab-ci.yml]: ✅ done

**Review by bobmshannon — DISMISSED** [review]

**Review by bobmshannon — DISMISSED** [review]

**Review by bobmshannon — APPROVED** [review]

**Pull request merged**